api/client/src/main/java/org/projectnessie/client/api/ns/ClientSideGetMultipleNamespaces.java

```diff
@@ -61,33 +60,24 @@ public GetNamespacesResponse get() throws NessieReferenceNotFoundException {
     try {
       GetEntriesBuilder getEntries = api.getEntries().refName(refName).hashOnRef(hashOnRef);
 
-      String filter = null;
+      String filter = "entry.contentType == 'NAMESPACE'";
       if (namespace != null && !namespace.isEmpty()) {
         String nsName = namespace.name();
-        filter =
+        filter +=
             onlyDirectChildren
                 ? format(
-                    "size(entry.keyElements) == %d && entry.encodedKey.startsWith('%s.')",
+                    " && size(entry.keyElements) == %d && entry.encodedKey.startsWith('%s.')",
                     namespace.getElementCount() + 1, nsName)
                 : format(
-                    "entry.encodedKey == '%s' || entry.encodedKey.startsWith('%s.')",
+                    "&& entry.encodedKey == '%s' || entry.encodedKey.startsWith('%s.')",
```

Comment:
Guess this one needs brackets:
```suggestion
                    "&& (entry.encodedKey == '%s' || entry.encodedKey.startsWith('%s.'))",
```